File: multilibrary/src/main/java/com/zhongjh/albumcamerarecorder/recorder/db/RecordingItem.java
package com.zhongjh.albumcamerarecorder.recorder.db;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * 音频文件的存储
 */
public class RecordingItem implements Parcelable {
    private String mName; // file name
    private String mFilePath; //file path
    private int mId; //id in database
    private int mLength; // length of recording in seconds
    private long mTime; // date/time of the recording

    public RecordingItem()
    {
    }

    public String getFilePath() {
        return mFilePath;
    }

    public void setFilePath(String filePath) {
        mFilePath = filePath;
    }

    public int getLength() {
        return mLength;
    }

    public void setLength(int length) {
        mLength = length;
    }

    public int getId() {
        return mId;
    }

    public void setId(int id) {
        mId = id;
    }

    public String getName() {
        return mName;
    }

    public void setName(String name) {
        mName = name;
    }

    public long getTime() {
        return mTime;
    }

    public void setTime(long time) {
        mTime = time;
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.mName);
        dest.writeString(this.mFilePath);
        dest.writeInt(this.mId);
        dest.writeInt(this.mLength);
        dest.writeLong(this.mTime);
    }

    protected RecordingItem(Parcel in) {
        this.mName = in.readString();
        this.mFilePath = in.readString();
        this.mId = in.readInt();
        this.mLength = in.readInt();
        this.mTime = in.readLong();
    }

    public static final Creator<RecordingItem> CREATOR = new Creator<RecordingItem>() {
        @Override
        public RecordingItem createFromParcel(Parcel source) {
            return new RecordingItem(source);
        }

        @Override
        public RecordingItem[] newArray(int size) {
            return new RecordingItem[size];
        }
    };
}

File: common/src/main/java/com/zhongjh/common/entity/RecordingItem.kt
package com.zhongjh.common.entity

import android.os.Parcel
import android.os.Parcelable

/**
 * 音频文件的存储
 * @author zhongjh
 */
class RecordingItem : LocalFile, Parcelable {

    /**
     * 网址
     */
    var url: String? = null

    constructor() : super()

    constructor(input: Parcel) : super(input) {
        url = input.readString()
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        super.writeToParcel(dest, flags)
        dest.writeString(url)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<RecordingItem> {
        override fun createFromParcel(parcel: Parcel): RecordingItem {
            return RecordingItem(parcel)
        }

        override fun newArray(size: Int): Array<RecordingItem?> {
            return arrayOfNulls(size)
        }
    }


}